### PR TITLE
add support for the ListSecrets restful API

### PR DESCRIPTION
**Pull request by wanghouqi**
this restful API will be exposed to public as:
 POST <ehsm_srv_address>/ehsm?Action=ListSecrets

 Queries all secrets created by your appid. Maximum 4000 line

 test: passed the python test

Signed-off-by: wanghouqi <wanghouqi@vip.qq.com>

### CODE REVIEW

File: test/cli/listSecrets.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--url', type=str, help='the address of the ehsm_kms_server', required=True)
+    parser.add_argument('--secretName', type=str, help='the name of the secret')
+    args = parser.parse_args()
+
+    base_url = args.url + "/ehsm?Action="
+    print(base_url)
+    return base_url, args.secretName
+
+def listSecrets(base_url, secretName):
```
Comment (HawieLong) [inline test/cli/listSecrets.py:24]: def listSecrets(base_url, secretName=None):, 默认不传就是None

File: test/cli/listSecrets.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    payload = OrderedDict()
+    if secretName != None:
+        payload["secretName"] = secretName
+    params = _utils_.init_params(payload)
```
Comment (HawieLong) [inline test/cli/listSecrets.py:28]: if secretName != None:
        payload["secretName"] = secretName
        params = _utils_.init_params(payload)
else:
        params = _utils_.init_params()

File: ehsm_kms_service/secret_manager_apis.js [hunk 3]
```diff
@@ -265,9 +265,64 @@ const createSecret = async (res, appid, payload, DB) => {
...
+        if (!checkStringParam(secretName, false, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+
+        // build selector for search secret_metadata
+        let selector = {
+            appid: appid
+        }
+        if (secretName != '' && secretName != undefined) {
```
Comment (HawieLong) [inline ehsm_kms_service/secret_manager_apis.js:282]: whether should query deleted flag? deletedflag only use for tag or can be used on secretname?

Comment (wanghouqi) [reply ehsm_kms_service/secret_manager_apis.js:282]: The function just return meta info, so logic delete secret should be return, and result has plannedDeleteTime

File: test/cli/listSecrets.py (hunk 1 above)
Comment (wanghouqi) [reply test/cli/listSecrets.py]: will be change

File: test/cli/listSecrets.py (hunk 2 above)
Comment (wanghouqi) [reply test/cli/listSecrets.py]: will be change

File: ehsm_kms_service/secret_manager_apis.js (hunk 3 above)
Comment (yang8621) [reply ehsm_kms_service/secret_manager_apis.js:296]: This is the question I want to know. we can note and discuss.

**Review by yang8621 — APPROVED** [review]

**Comment by wanghouqi** [comment]
Move to https://github.com/intel/ehsm/pull/155

**Comment by wanghouqi** [comment]
Move to https://github.com/intel/ehsm/pull/155

**Pull request closed without merging**